app/assets/javascripts/components/cloud_subnet/cloud-subnet-form.js

```diff
@@ -39,7 +39,32 @@ function cloudSubnetFormController(API, miqService) {
       }).catch(miqService.handleFailure);
     } else {
       API.get('/api/cloud_subnets/' + vm.cloudSubnetFormId + '?expand=resources&attributes=ext_management_system.name,cloud_tenant.name,cloud_network.name').then(function(data) {
-        Object.assign(vm.cloudSubnetModel, _.pick(data, 'name', 'ext_management_system', 'cloud_network', 'cloud_tenant', 'network_protocol', 'cidr', 'dhcp_enabled', 'gateway'));
+        Object.assign(vm.cloudSubnetModel, _.pick(data, 'name', 'ext_management_system', 'cloud_network', 'cloud_tenant', 'network_protocol', 'cidr', 'dhcp_enabled', 'gateway', 'extra_attributes', 'dns_nameservers'));
+        vm.cloudSubnetModel.allocation_pools = "";
+        vm.cloudSubnetModel.host_routes = "";
+        if (vm.cloudSubnetModel.extra_attributes !== undefined) {
+          if (vm.cloudSubnetModel.extra_attributes.allocation_pools !== undefined) {
+            vm.cloudSubnetModel.extra_attributes.allocation_pools.forEach(function(val, i) {
+              if (i > 0) {
+                vm.cloudSubnetModel.allocation_pools += "\n";
+              }
+              vm.cloudSubnetModel.allocation_pools += val["start"] + "," + val["end"];
+            });
+          }
+          if (vm.cloudSubnetModel.extra_attributes.host_routes !== undefined) {
+            vm.cloudSubnetModel.extra_attributes.host_routes.forEach(function(val, i) {
+              if (i > 0) {
+                vm.cloudSubnetModel.host_routes += "\n";
+              }
+              vm.cloudSubnetModel.host_routes += val["destination"] + "," + val["nexthop"];
+            });
+          }
+        }
+        if (vm.cloudSubnetModel.dns_nameservers !== undefined) {
+          vm.cloudSubnetModel.dns_nameservers = vm.cloudSubnetModel.dns_nameservers.join("\n");
+        } else {
+          vm.cloudSubnetModel.dns_nameservers = ""
```

Comment:
Why not move this from `else` to `vm.cloudSubnetModel.allocation_pools = "";vm.cloudSubnetModel.host_routes = "";` block? It would be more readable :)

Reply:
And missing `;` :)

Reply:
Good catch. I wish javacsript would just require (or forbid) the semicolons. They're too easy to forget coming from a mostly ruby dev environment, especially when the code works just fine without them. I'll move this bit up and add the semicolon.

Reply:
Actually, now I remember why I didn't put that higher up. Unlike the other two cases, where we're pulling the model params from a different parameter (from extra_attributes), for dns_nameservers, we're actually modifying the value we get from the api and using the same parameter to store the result. If I set this to "" above, then I will have lost the source value. At this point, if the value is defined, the code inserts the newlines as needed, otherwise, it sets it to "". I will update this with the semicolon added at the end of the assignment line.

Reply:
That sounds good 👍